themes/ssp/core/no_state.tpl.php

```diff
@@ -1,15 +1,16 @@
 <?php
 
-echo('<h2>' . $this->t('{core:no_state:suggestions}') . '</h2>');
-echo('<ul class="list-unstyled">');
-echo('<li>' . $this->t('{core:no_state:suggestion_goback}') . '</li>');
-echo('<li>' . $this->t('{core:no_state:suggestion_closebrowser}') . '</li>');
-echo('</ul>');
-
-echo('<h3>' . $this->t('{core:no_state:causes}') . '</h3>');
-echo('<ul class="list-unstyled">');
-echo('<li>' . $this->t('{core:no_state:cause_backforward}') . '</li>');
-echo('<li>' . $this->t('{core:no_state:cause_openbrowser}') . '</li>');
-echo('<li>' . $this->t('{core:no_state:cause_nocookie}') . '</li>');
-echo('</ul>');
+echo ('<h2>' . $this->t('{core:no_state:suggestions}') . '</h2>');
+echo ('<ul class="list-unstyled">');
+echo ('<li>' . $this->t('{core:no_state:suggestion_badlink}') . '</li>');
+echo ('<li>' . $this->t('{core:no_state:suggestion_goback}') . '</li>');
+echo ('<li>' . $this->t('{core:no_state:suggestion_closebrowser}') . '</li>');
+echo ('</ul>');
 
```

Comment:
why not remove echo from the entire file and move to an html file with 
`<?= $this->t('{core:no_state:suggestions}') ; ?>`
It will be much more readable.

Reply:
Resolved by commit [Remove sequentially echos ](https://github.com/EGI-Federation/simplesamlphp-module-themeegi/pull/14/commits/0d8e00ed36955b6ffb716baef99a40a281115645)